### ISSUE TITLE
Arabidopsis proteome sequences comparison

### ISSUE THREAD

**Issue by edeutsch** [PlantProteomes/SeqComparison#5]
Hi everyone, once you get your sequence comparison programs working, please download the latest Arabidopsis sequences
(see commit https://github.com/PlantProteomes/SeqComparison/commit/6b1a327f5b4d3cf44e294367a984d49e3e131d4c)
You should have this file after a git pull
Please done commit the data files you pull down. They're too big.

and then generate a table that looks something like this:
![image](https://user-images.githubusercontent.com/12707718/148862789-a9440ce3-40f1-4d35-9542-a0281eedb3e9.png)

with all the relevant numbers filled in.

Unsure about something? Ask questions!

**Comment by cashewballerz** [comment]
is there a general timeline for when you want this completed? Additionally form the table itself it appears you want us to basically compare 5 datasets at once is what I'm understanding and put it into a table format. It would be what we have done on the Maize files, but at a larger scale and an actual visual with numbers.

**Comment by edeutsch** [comment]
Yes! Let's aim for the end of the week? I think that should be doable.

Most of the comparisons are pairwise, not all together. Only the very bottom line is all together. But if you're creative, you can compute that too without developing anything different that what we've been discussing.

And there are 8 instead of 5?

**Comment by cashewballerz** [comment]
Yup 8, I saw the three columns labeled sequence, distinct and unique, and subtracted from 8 and assumed 5 when there are eight datasets themselves, to begin with. So you'd you than like us actually to take 8 inputs as files, or would you want to be in someway flexible such that given an X amount of files, we can create a table off than number so say three files, we would then make a 3x3 table with information printed. If it were to be like that, it would likely take me longer since my current accepted input is only 2 files. I would have to change the structure of my program to accept any amount of files, so I assume the best way would be to take the input of every file and take the count to form the table. However, I'm not exactly sure how to hardcode that in.

**Comment by edeutsch** [comment]
So the ultimate fancy program would take 8 files and generate the table as designed. But don't aim for the ultimate fancy program if your current pairwise program doesn't fully work yet. First get your pairwise program working and fully understood and tidied up. Then you could fill in the table by running each pair manually and filling in the table in an Excel spreadsheet as you go. Once you feel confident that you can do that, then you can try aiming for the fancy program.

**Comment by cashewballerz** [comment]
What's the difference between distinct and unique here? I have my code identifying unique sequences but distinct is just a synonym.

**Comment by cashewballerz** [comment]
Also the do the shaded in squared not be filled?

**Comment by edeutsch** [comment]
correct!

**Comment by cashewballerz** [comment]
I have an idea by using pandas at the class level to basically update the data frame itself with every file that is inputted. I'm coming across 2 errors however, and I've spent a bit of time attempting to think of solutions. First is the fact that how can I transfer the date properly since I've created the pandas data frame at the class level, creating a new object should suffice in pulling it. The other is what function can I use to get the number counts of the data itself rather than the raw data, I tried Len but it appears I was running into multiple errors. Any help would be awesome! If any of you have solutions, please tell me during Thursday's meeting as I continue to poke through this. The most recent version with errors is put into GitHub for viewing.

**Comment by edeutsch** [comment]
I think you should put some print statements in there to get a good look at what the variables you're trying to use contain.

What do you want your dataframe to look like? What are the rows and columns?

I think what's causing your problems is line 77:
```
        df = pd.DataFrame(self.stats)
```
You're getting a dataframe of something but I don't think it resembles what you want.

I think you would be best off creating your dataframe first (with just zeros) getting all the rows and columns right so you can do print(df) and see a nice template.

And *then* start filling in the values that you compute.

Does that seem sensible?

**Comment by cashewballerz** [comment]
![image](https://user-images.githubusercontent.com/34139922/152250784-1e9881e1-a21a-4773-8eb4-4b5bcc3732d7.png)
Progress is being made. Now I just have to be able to multiply it across different levels of files, which I assume I'll need a lot of loops for. I'm unsure if I'll be able to make the meeting tomorrow, so feel free to tinker around with my code and add suggestions and potential code to push along, that would be awesome. As for the other recent task you gave I'm working on it with the new update you gave in the other issues tab

**Comment by edeutsch** [comment]
okay, sounds good. I will be at the usual Zoom room Thursday at 5 if you're able to be there.

**Comment by cashewballerz** [comment]
Sorry I couldn't make the meeting today. Do you think I could meet with you Monday at 330 to do a progress check and just get some general help on the work we are doing?

**Comment by edeutsch** [comment]
sure thing, I'll be there 3:30 in the usual place. See you then!
